maa-cli/src/run/preset/reclamation.rs

```diff
@@ -0,0 +1,143 @@
+use super::MAAValue;
+
+#[repr(u8)]
+#[cfg_attr(test, derive(Debug, PartialEq))]
+#[derive(Clone, Copy)]
+enum Theme {
+    Tales,
+}
+
+impl Theme {
+    const fn to_str(self) -> &'static str {
+        match self {
+            Theme::Tales => "Tales",
+        }
+    }
+}
+
+impl clap::ValueEnum for Theme {
+    fn value_variants<'a>() -> &'a [Self] {
+        &[Theme::Tales]
+    }
+
+    fn to_possible_value(&self) -> Option<clap::builder::PossibleValue> {
+        Some(clap::builder::PossibleValue::new(self.to_str()))
+    }
+}
+
+#[derive(clap::Args)]
+pub struct ReclamationParams {
+    /// Theme of the reclamation algorithm
+    theme: Theme,
+    /// Mode of the reclamation algorithm
+    ///
+    /// 0: build and restart again and again. **This mode will discard all the progress**.  
+    /// 1: craft some products and fall back to the previous checkpoint.
+    #[arg(short = 'm', long, default_value = "1", verbatim_doc_comment)]
+    mode: i32,
+    /// Tool to craft in mode 1
+    #[arg(short = 'C', long)]
+    tool_to_craft: Option<String>,
+    /// Mode to increase the number of tools to craft
+    ///
+    /// 0: increase the number by clicking the button.
+    /// 1: increase the number by holding the button.
+    #[arg(long, default_value = "0", verbatim_doc_comment)]
+    increase_mode: i32,
+    /// Number of craft each batch
+    #[arg(long, default_value = "16")]
+    num_craft_batches: i32,
+}
```

Comment:
@Alan-Charred 老师，麻烦看看生息演算的参数和 help 有没有问题。

Reply:
内容完全没有问题！感谢！
这个注释是可以直接作为help的文字吗？我脑海中是以下文字，您看着改改。（我对rust不太熟悉，也不懂输出结构好不好看。

话说，是不是`increase_mode` 和 `num_craft_batches`应该像`tool_to_craft`一样有个option？哦它们有默认值。
如果可以，还请后续加一个检查，如果`tool_to_craft`为空的话，报个错。
我们其实是希望能给它一个“荧光棒”作为默认值，但是我不确定 cli 是否要支持多语言。您看看怎么办比较好？

```Rust
    /// Reclamation Algorithm theme
    ///
    /// Tales: Tales Within the Sand
    #[arg(short = 't', long, default_value = "Tales", verbatim_doc_comment)]
    theme: String,
    /// Reclamation Algorithm task mode
    ///
    /// 0: farm prosperity points by repeatedly entering and exiting stages.
    ///     **This task mode should only be used when no save exists**.
    ///     **Using it with an active save may result in losing your progress**.
    /// 1: farm prosperity points by crafting tools.
    ///     **This task mode should only be used when you already have a save and can craft certain tools**.
    ///     **Using it may result in losing your progress after last calculation day**.
    ///     **It is recommend to start task from a new calculation day**.
    #[arg(short = 'm', long, default_value = "1", verbatim_doc_comment)]
    mode: i32,
    /// Name of tool to craft in mode 1
    #[arg(short = 'C', long)]
    tool_to_craft: Option<String>,
    /// Method to interactive with the add button when increasing the crafting quantity
    ///
    /// 0: increase the number by clicking the button.
    /// 1: increase the number by holding the button.
    #[arg(long, default_value = "0", verbatim_doc_comment)]
    increase_mode: i32,
    /// Number of batches in each game run, with each batch containing 99 items
    #[arg(long, default_value = "16")]
    num_craft_batches: i32,
```

草好像对theme的描写有点问题，我得去改改文档。

Reply:
感谢，不过这个 help 好像暂时不支持用 `**` 来加粗，我来研究一下怎么给 help 加上格式。另外 theme 不用 String 而是 ValueEnum, 这样可以生成补全和可选项。